client/lib/plans/index.js

```diff
@@ -16,27 +16,34 @@ import {
 	FEATURES_LIST,
 	PLANS_LIST,
 	PLAN_FREE,
-	PLAN_JETPACK_BUSINESS,
-	PLAN_JETPACK_BUSINESS_MONTHLY,
 	PLAN_JETPACK_FREE,
-	PLAN_JETPACK_PERSONAL,
-	PLAN_JETPACK_PERSONAL_MONTHLY,
-	PLAN_JETPACK_PREMIUM,
-	PLAN_JETPACK_PREMIUM_MONTHLY,
 	PLAN_PERSONAL,
 } from 'lib/plans/constants';
+import {
+	TERM_ANNUALLY,
+	TERM_MONTHLY,
+	TYPE_BUSINESS,
+	TYPE_FREE,
+	TYPE_PERSONAL,
+	TYPE_PREMIUM,
+} from './constants';
 
 /**
  * Module vars
  */
 const isPersonalPlanEnabled = isEnabled( 'plans/personal-plan' );
 
-export function isFreePlan( plan ) {
-	return plan === PLAN_FREE || plan === PLAN_JETPACK_FREE;
+export function getPlans() {
+	return PLANS_LIST;
 }
 
-export function getPlan( plan ) {
-	return PLANS_LIST[ plan ];
+export function getPlan( planKey ) {
+	if ( Object.prototype.toString.apply( planKey ) === '[object Object]' ) {
+		if ( values( PLANS_LIST ).indexOf( planKey ) !== -1 ) {
+			return planKey;
+		}
+	}
+	return PLANS_LIST[ planKey ];
```

Comment:
how about something like this…

```js
/**
 * Returns plan information given name of plan
 *
 * @param {string} planKey name of plan (@see ./constants)
 * @returns {?object} plan information if available
 */
export function getPlan( planKey ) {
  if ( 'development' === process.env.NODE_ENV ) {
    if ( 'string' !== typeof planKey ) {
      console.error( 'Received a non-string plan name for getPlan(): please check call-site' );
      if ( planKey.slugOrWhatever ) {
        console.error( 'Did you mean to pass `planKey.slugOrWhatever` instead?' );
      }
    }

    return PLANS_LIST[ planKey ];
  }
}
```

Reply:
As I mentioned above, we can probably get rid of the part for handling Object argument. I like this here, it's pretty much a static typing :-) I wonder how easy would it be to start using https://flow.org/ for cases like this here.

Reply:
>  it's pretty much a static typing

well, if by "pretty much" we mean "the polar opposite" haha. why have the computer automate bug-finding for us at no cost to our customers if we can do runtime type-checking manually and incompletely by only adding development time?

> I wonder how easy would it be to start using https://flow.org/ for cases like this here.

super easy technically and complicated holistically. there are a few of us who have been exploring gradually-typed systems for coming up on a year almost. if you are interested then please join the discussion in the framework channel on Slack 🙂